lib/ASTGen/Sources/ASTGen/Decls.swift

```diff
@@ -75,43 +99,42 @@ extension ASTGenVisitor {
     } else {
       firstName = nil
     }
+    let (secondName, secondNameLoc) = node.secondName.bridgedIdentifierAndSourceLoc(in: self)
 
-    let secondName = node.secondName.bridgedIdentifier(in: self)
-
-    let type = visit(node.type).rawValue
-
-    return .decl(ParamDecl_create(ctx, loc, loc, firstName, loc, secondName, type, declContext))
+    return .decl(
+      ParamDecl_create(
+        self.ctx,
+        self.declContext,
+        specifierLoc,
+        firstName,
+        self.bridgedSourceLoc(for: node.firstName),
+        secondName,
+        secondNameLoc,
+        self.visit(node.type).rawValue
+      )
+    )
   }
 
   public func visit(_ node: FunctionDeclSyntax) -> ASTNode {
-    let staticLoc = bridgedSourceLoc(for: node)
-    let funcLoc = bridgedSourceLoc(for: node.funcKeyword)
-    let (name, nameLoc) = node.name.bridgedIdentifierAndSourceLoc(in: self)
+    // FIXME: Compute this location
+    let staticLoc: BridgedSourceLoc = nil
```

Comment:
Gah we really need to turn the staticness into a decl attribute, I started working on that a little while back but never got the time to finish it